instrumentation/jms-1.1/src/jms2Test/groovy/JMS2Test.groovy

```diff
@@ -192,12 +190,14 @@ class JMS2Test extends AgentTestRunner {
       trace(0, 1) { // Consumer trace
         span(0) {
           parent()
-          operationName destinationType + "/" + destinationName + " receive"
-          spanKind CLIENT
+          operationName destinationName + " receive"
+          spanKind CONSUMER
```

Comment:
IMO this is very confusing right now: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md#span-kind

I read convention like "CONSUMER if async, CLIENT if sync and sends response back"

Reply:
I think we need to clarify, because I think this change breaks this use case:

* A servlet request reads a message from a queue

I would expect for the span "reading from queue" to be parented to the span "servlet request" (with a link to the "producer" span).

With this change, the span "reading from queue" is parented to the "producer" span, and is no longer in the span tree of the "servlet request".

Reply:
Setting aside that this is quite an unusual use-case for me :), I say that messaging semantic convention does not support this anyway. You see, `CLIENT` is a variant of `PRODUCER`, it means an operation creating a new message. Thus `receive` operation cannot be `CLIENT`. It can be either `CONSUMER` or `SERVER` :)

And I think you are mixing two separate things here: setting span's `kind` and parent/link. They are mostly independent from each other.

Reply:
This PR changes both the span kind and the parent/link behavior. I disagree with both changes 😂 but am very likely wrong about one or more likely both. I added to tomorrow's [agenda](https://docs.google.com/document/d/1WK9h4p55p8ZjPkxO75-ApI9m0wfea6ENZmMoFRvXSCw), and in case there's still disagreement after that, we can merge as is and escalate spec clarification.